public/pages/DefineDetector/containers/DefineDetector.tsx

```diff
@@ -218,6 +262,61 @@ export const DefineDetector = (props: DefineDetectorProps) => {
     }
   };
 
+  const handleDataSourceChange = ([event]) => {
+    const dataSourceId = event?.id;
+    if (!dataSourceId) {
+      getNotifications().toasts.addDanger(
+        prettifyErrorMessage('Unable to set data source.')
+      );
+    } else {
+      setMDSCreateState({
+        queryParams: dataSourceId,
+        selectedDataSourceId: dataSourceId,
+      });
+    }
+  };
+
+  let renderDataSourceComponent: ReactElement | null = null;
+  if (dataSourceEnabled) {
+    if (props.isEdit) {
+      const DataSourceMenu =
+      getDataSourceManagementPlugin()?.ui.getDataSourceMenu<DataSourceViewConfig>();
+    renderDataSourceComponent = (
+      <DataSourceMenu
+        setMenuMountPoint={props.setActionMenu}
+        componentType={'DataSourceView'}
+        componentConfig={{
+          activeOption: [{ id: dataSourceId }],
+          fullWidth: false,
+          savedObjects: getSavedObjectsClient(),
+          notifications: getNotifications(),
+        }}
+      />
+    );
+    } else {
+      const DataSourceMenu =
+        getDataSourceManagementPlugin().ui.getDataSourceMenu<DataSourceSelectableConfig>();
+      renderDataSourceComponent = useMemo(() => {
+        return (
+          <DataSourceMenu
+            setMenuMountPoint={props.setActionMenu}
+            componentType={'DataSourceSelectable'}
+            componentConfig={{
+              fullWidth: false,
+              activeOption: [{ id: MDSCreateState.selectedDataSourceId }],
```

Comment:
How are we handling the case where `MDSCreateState.selectedDataSourceId = ''` but `Local cluster` is disabled? When this is initially defined, the default value is `''` which means the initial selected option will be the `Local cluster`. I think a potential solution here is to set the initial state of  `MDSCreateState.selectedDataSourceId = undefined` and something like the following:

```Typescript
activeOption: MDSCreateState.selectedDataSourceId === undefined ? [{ id: MDSCreateState.selectedDataSourceId }] : undefined
```

This enables the `DataSourceSelectable` component to handle the default selected choice. Could you please check for other pages as well?

Reply:
The case I'm referring to is how you set the initial render of the page. Currently, if `Local cluster` is disabled and you navigate to the `/app/anomaly-detection-dashboards#/`, it will throw the toast message every time which should be avoided. This is because `MDSCreateState.selectedDataSourceId = ''` when this is not valid.

From datasource UI component side, it is not possible for a user to select `Local cluster` because the component will check to see if `Local cluster` is disabled and add/omit it if necessary. Of course, the toast should still be thrown if the user intentionally/unintentionally sets the url to `dataSourceId=` because it's invalid.

Reply:
ic, this is the issue that AD and IMS have in the future playground, right? Let's address it in the next pr, I haven't pull latest change from OSD in this pr. So the bug has not been exposed in my local yet

Reply:
From MDS side, here should be the expected behaviors:

If `Local cluster` is enabled:
- The initial render of the page should have `defaultDataSource` as the selected option, not the `Local cluster`. This can be easily done if `activeOptions = undefined`. The url should reflect this choice with `?dataSourceId=<defaultDataSource>`
- The user should be able to select `Local cluster` without any Toast message being shown (and the url should reflect this choice with `?dataSourceId=`)
- The `?dataSourceId=` should be a valid url and this should represent the `Local cluster` being selected

If `Local cluster` is disabled:
- The initial render of the page should have `defaultDataSource` as the selected option, not the `Local cluster`. This can be easily done if `activeOptions = undefined`. The url should reflect this choice with `?dataSourceId=<defaultDataSource>`
- The user shouldn't be able to select `Local cluster` from UI component (this is datasource UI responsibility; this should already be covered)
- The `?dataSourceId=` can either do one of 2 things:
    - Throw the toast message because the `Local cluster` is attempted to be made but is not valid
    - **(Preferred)**: redirect the url to the `?dataSourceId=<defaultDataSource>` and put the `defaultDatasource` as the selected option. If this is non-trivial, this is the preferred option

Reply:
thanks for the details! I will do a sanity check on all the data source picker components and make adjustment in the next pr. Wanted to focus on adding MDS support for creation&update flow in this pr

Reply:
@kgcreative I may need confirmation if this makes sense from UX perspective. For right now, on datasource UI side, `defaultDataSource` is prioritized over `Local cluster` so we shouldn't change these priorities.